src/sentry/migrations/0049_auto_20200302_1955.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.28 on 2020-03-02 18:25
+from __future__ import unicode_literals
+
+from datetime import datetime
+from django.db import migrations
+from django.db.models import OuterRef, Exists
+
+
+def mark_all_projects_completed_alert_rule_onboarding(apps, schema_editor):
+    from sentry.models import OnboardingTaskStatus, OnboardingTask
+
+    Project = apps.get_model("sentry", "project")
+    OrganizationOnboardingTask = apps.get_model("sentry", "organizationonboardingtask")
+
+    # Locate all projects, annotated which have not completed alert onboarding
+    # tasks.
+    #
+    # Queryplan:
+    #
+    #   Seq Scan on sentry_project (cost=0.00..24836341662.11 rows=297362 width=103)
+    #   Filter: ((date_added <= '2020-03-02 00:00:00+00'::timestamp with time zone) AND (NOT (alternatives: SubPlan 3 or hashed SubPlan 4)))
+    #   SubPlan 1
+    #   -> Seq Scan on sentry_organizationonboardingtask u0 (cost=0.00..27838.89 rows=1 width=0)
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id) AND (task = 10))
+    #   SubPlan 2
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_1 (cost=0.00..25585.33 rows=20264 width=8)
+    #   Filter: ((status = 1) AND (task = 10))
+    #   SubPlan 3
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_2 (cost=0.00..27838.89 rows=1 width=0)
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id) AND (task = 10))
+    #   SubPlan 4
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_3 (cost=0.00..25585.33 rows=20264 width=8)
+    #   Filter: ((status = 1) AND (task = 10))
+    target_projects = Project.objects.annotate(
+        alert_task_complete=Exists(
+            OrganizationOnboardingTask.objects.filter(
+                organization_id=OuterRef("organization_id"),
+                project_id=OuterRef("id"),
+                task=OnboardingTask.ALERT_RULE,
+                status=OnboardingTaskStatus.COMPLETE,
+            )
+        )
+    ).filter(alert_task_complete=False)
+
+    for project in target_projects:
```

Comment:
We don't we have a querysetwrapper here?

Reply:
I actually removed it because of Dan's comment here https://github.com/getsentry/sentry/pull/17411#discussion_r386673215

Reply:
Okay, see updated comment here https://github.com/getsentry/sentry/pull/17411#discussion_r386747192